internal/list.go

```diff
@@ -11,3 +32,210 @@ func newListProxy(client *HazelcastClient, serviceName *string, name *string) (*
 	}
 	return &ListProxy{parSpecProxy}, nil
 }
+
+func (list *ListProxy) Add(element interface{}) (changed bool, err error) {
+	elementData, err := list.validateAndSerialize(element)
```

Comment:
This validateAndX methods used by all proxy types so we need to make their internal error texts more generic.
For example this `Add` method will return an error for `NIL_KEY_IS_NOT_ALLOWED` .

Reply:
@asimarslan Do you have any suggestions? Can we use something like `NIL_ARG_IS_NOT_ALLOWED`?